pkgs/development/tools/rust/cargo-generate/default.nix

```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchFromGitHub, rustPlatform, Security, openssl, pkgconfig, libiconv, curl }:
+
+rustPlatform.buildRustPackage rec {
+  name = "cargo-generate-${version}";
+  version = "0.3.0";
+
+  src = fetchFromGitHub {
+    owner = "ashleygwilliams";
+    repo = "cargo-generate";
+    rev = "v${version}";
+    sha256 = "0n6na6xq4bvs9hc7vc86qqmlrkv824qdmja27b21l2wz3l77r4jb";
+  };
+
+  cargoSha256 = "00fgzh1s63rr1vs3ahra604m81fc4imx3s09brw2y0n46syhwypi";
+
+  nativeBuildInputs = [ pkgconfig ];
+
+  buildInputs = [ openssl libiconv curl ]
+    ++ stdenv.lib.optional stdenv.isDarwin Security;
+
+  # Tests depend on a valid git configuration.
+  # Rather than doing some crazy hack and adding additional dependencies
+  # we simply avoid them completely.
+  doCheck = false;
```

Comment:
Maybe try
```
preCheck = ''
  export GIT_COMMITTER_NAME=nixbld
  export EMAIL=nixbld@localhost
  export GIT_COMMITTER_DATE=$SOURCE_DATE_EPOCH
'';
```

Reply:
I think the whole test suit will do quite some side-effects. Just adding this produces following error during build

```
---- basics::it_substitutes_projectname_in_cargo_toml stdout ----
thread 'basics::it_substitutes_projectname_in_cargo_toml' panicked at 'Unexpected failure.
code-128
stderr=```fatal: invalid date format: 1
```
command=`"git" "commit" "--message" "initial commit"`
code=128
stdout=``````
stderr=```fatal: invalid date format: 1
```
', /build/cargo-generate-0.3.0-vendor/assert_cmd/src/assert.rs:140:13
```

I'm not sure how to fix this but I can try to figure it out. Anyway I would expect there will be more issues along the way and am not sure if it's worth trying to fix all of those.